app-arch/brotli/brotli-1.0.2.recipe

```diff
@@ -0,0 +1,149 @@
+SUMMARY="A generic-purpose lossless compression algorithm"
+DESCRIPTION="A generic-purpose lossless compression algorithm that \
+compresses data using a combination of a modern variant of the LZ77 algorithm, \
+Huffman coding and 2nd order context modeling, with a compression ratio \
+comparable to the best currently available general-purpose compression methods. \
+It is similar in speed with deflate but offers more dense compression."
+HOMEPAGE="https://github.com/google/brotli"
+COPYRIGHT="2009, 2010, 2013-2016 Brotli Authors"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/google/brotli/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="c2cf2a16646b44771a4109bb21218c8e2d952babb827796eb8a800c1f94b7422"
+
+ARCHITECTURES="x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+# For now the library will follow $portVersion,
+# but upstream said that will change in the future.
+libVersion="1.0.2"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	brotli$secondaryArchSuffix = $portVersion
+	cmd:brotli$secondaryArchSuffix = $portVersion
+	lib:libbrotlicommon$secondaryArchSuffix = $libVersionCompat
+	lib:libbrotlidec$secondaryArchSuffix = $libVersionCompat
+	lib:libbrotlienc$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	brotli${secondaryArchSuffix}_devel = $portVersion
+	devel:libbrotlicommon$secondaryArchSuffix = $libVersionCompat
+	devel:libbrotlicommon_static$secondaryArchSuffix = $libVersion
+	devel:libbrotlidec$secondaryArchSuffix = $libVersionCompat
+	devel:libbrotlidec_static$secondaryArchSuffix = $libVersion
+	devel:libbrotlienc$secondaryArchSuffix = $libVersionCompat
+	devel:libbrotlienc_static$secondaryArchSuffix = $libVersion
+	"
+REQUIRES_devel="
+	brotli$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:g++$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	"
+
+PYTHON_PACKAGES=(python python3)
+PYTHON_VERSIONS=(2.7 3.6)
+# We don't have python2 for secondaryArch,
+# and gcc2 doesn't support the flags passed by python3
+if [ -n "$secondaryArchSuffix" ]; then
+	unset PYTHON_PACKAGES[0]
+	unset PYTHON_VERSIONS[0]
+elif [ "$effectiveTargetArchitecture" = "x86_gcc2" ]; then
+	unset PYTHON_PACKAGES[1]
+	unset PYTHON_VERSIONS[1]
+fi
+for i in "${!PYTHON_PACKAGES[@]}"; do
+pythonPackage=${PYTHON_PACKAGES[i]}
+pythonVersion=${PYTHON_VERSIONS[$i]}
+eval "PROVIDES_${pythonPackage}=\"\
+	${portName}_$pythonPackage = $portVersion\
+	\"; \
+REQUIRES_$pythonPackage=\"\
+	haiku\n\
```

Comment:
brotli_python{,3} should also depend on 
```
		brotli$secondaryArchSuffix == $portVersion base
```
Hmm, since this is exactly REQUIRES_devel, we can fix the dependencies of brotli_python{,3} with the following change:
```diff
 REQUIRES_$pythonPackage=\"\
-	haiku\n\
+	$REQUIRES_devel \
 	cmd:python$pythonVersion\
 	\""
```

Well, I don't know if it is desirable to _use_ `$REQUIRES_devel`, so please feel free to use whatever alternative you wish.

Reply:
It wouldn't make much sense to use `$REQUIRES_devel`, looking at the context, it would looks like those are some kind of devel package (they somewhat are lol).

Do we really need to make them depends on the base package? They're pretty much self-contained themselves

Reply:
I might be wrong, but I'm quite sure that if you install brotli_python without brotli then you cannot use it.
If this is correct then the package must declare this dependency.
Now, in the suggestion I made, I used `$REQUIRES_devel` because it was easier to write than to copy the exact same string that was used to define REQUIRES_devel.
Of course, you are free to use any construct you wish, as long as the package has this dependency.

Reply:
You can. The package have an version of brotli within itself (`_brotli.so`)

Reply:
Hmm, you are perfectly right.
So this means the dependencies are correct.
I think everything is OK then. I guess we can merge.
> Bumped copyright year

Cool! I'll merge soon.